xbmc/filesystem/CurlFile.h

```diff
@@ -61,10 +61,8 @@ namespace XFILE
       bool ReadString(char *szLine, int iLineLength) override { return m_state->ReadString(szLine, iLineLength); }
       ssize_t Read(void* lpBuf, size_t uiBufSize) override { return m_state->Read(lpBuf, uiBufSize); }
       ssize_t Write(const void* lpBuf, size_t uiBufSize) override;
-      virtual std::string GetMimeType() { return m_state->m_httpheader.GetMimeType(); }
-      std::string GetContent() override { return m_state->m_httpheader.GetValue("content-type"); }
+      virtual const std::string GetProperty(XFILE::FileProperty type, const std::string &name = StringUtils::Empty) const override;
```

Comment:
`const std::string &name = ""`? Simpler to read and no new (!) dependency of this header file to "StringUtils.h" (which is currently missing, btw, if you want to stick with `const std::string &name = StringUtils::Empty`).

Reply:
... and no need for `virtual` if you add `override`. I would remove the superflous `virtual`.